### PR TITLE
ieee802154: mcr20a cleanups (fixup CS handling & remove dead Kconfig bits)

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: drivers/ieee802154/ieee802154_mcr20a.h [hunk 1]
```diff
@@ -25,9 +25,7 @@ struct mcr20a_context {
 	struct gpio_callback irqb_cb;
 	struct device *spi;
 	struct spi_config spi_cfg;
-#if defined(CONFIG_IEEE802154_MCR20A_GPIO_SPI_CS)
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_mcr20a.h]: Make it build if only DT_NXP_MCR20A_0_CS_GPIO_CONTROLLER is present, no?

Comment (galak) [reply drivers/ieee802154/ieee802154_mcr20a.h]: I was torn on this, I removed it if we ever support multiple instances of the driver.

Comment (tbursztyka) [reply drivers/ieee802154/ieee802154_mcr20a.h]: keep the ifdef, let's handle multiple instances another time. (when we'll have proper device generation DT based)

Comment (galak) [reply drivers/ieee802154/ieee802154_mcr20a.h]: Done.

**Review by tbursztyka — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by avisconti — APPROVED** [review]
seems fine

**Pull request merged**